src/server/common_services/auth_server.js

```diff
@@ -496,6 +496,10 @@ function _prepare_auth_request(req) {
     req.has_bucket_action_permission = async function(bucket, action, bucket_path, req_query) {
         return has_bucket_action_permission(bucket, req.account, action, req_query, bucket_path);
     };
+
+    req.has_list_bucket_permission = function(bucket) {
```

Comment:
I still see `list_bucket` where you mean `list_buckets`.
I think it is confusing because:
- List bucket = list objects in the buckets.
- List buckets = list buckets under the account.